Lcam 225 check cc details

AafAnsari opened the pull request:
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-225)

- Added crown court outcome validation, exception handler, unit tests
- Update json schema with outcome fields

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

venkat980 commented:
The test coverage can be improved and the code smells need to be addressed.

venkat980 commented on crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java:
```diff
@@ -0,0 +1,45 @@
+        if (apiCrownCourtOutcome.isPresent()) {
+            final CrownCourtOutcome crownCourtOutcome = apiCrownCourtOutcome.get().getOutcome();
+            log.info("Validating crown court outcome " + crownCourtOutcome);
+            if (crownCourtOutcome != null) {
```
the two if statements can be merged here.

venkat980 commented on crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    @Test
+    public void givenCrownCourtOutcomeConvictedWithNullDateAndNullImprisoned_whenValidateIsInvoked_thenValidationFails() {
```
please remove the public modifier from the test methods

AafAnsari replied:
an oversight. I'll remove these

venkat980 commented on crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java:
```diff
@@ -0,0 +1,104 @@
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+public class CrownCourtDetailsValidatorTest {
```
please remove this public modifier

venkat980 reviewed (commented):
The test coverage can be improved.

AafAnsari commented:
> 

I have 100% code coverage for the validator class

AafAnsari commented:
> > > 
> > 
> > 
> > I have 100% code coverage for the validator class
> 
> if (apiCrownCourtOutcome.isPresent() && apiCrownCourtOutcome.get().getOutcome() != null) { One more condition needs to be covered in the Validator class. Can you please check?

I have removed this check and added the outcome field as required in the schema
